code/clue.js

```diff
@@ -1,59 +1,396 @@
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  if (!killerIsPicked) {
+    document.getElementById('killerLoader').style.visibility='visible' 
+    setTimeout(function () {
+      document.getElementById('killerLoader').style.visibility='hidden'
+      //excecutes shuffleWeapons-function above and makes favorite weapon random:
+      shuffleWeapons();
+      //randomly select a killer from the suspects. And add that to the mystery object.
+      mystery.killer = randomSelector(suspects); //mystery.killer selects the property 'killer' in mystery object and sets this equal to a randomly selected suspect. 
+      document.getElementById('killerCard').style.background = mystery.killer.color
+      document.getElementById('killerName').innerHTML = `Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`
+      document.getElementById('killerAge').innerHTML = `Age: ${mystery.killer.age}`
+      document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`
+      document.getElementById('killerWeapon').innerHTML = `Favorite weapon: ${mystery.killer.favorite}`
+      document.getElementById('killerDescription').innerHTML = `Description: ${mystery.killer.description}`
+      document.getElementById('killerImage').src = `${mystery.killer.image}`
+      killerIsPicked = true;
+    }, 2500)
+  }
+};
+//Added eventlistener below instead of "onclick" in HTML. Don't know which way is the best way to do it. This only works if the eventlistener is outside of the function. 
+//First param: type of event. Second param: the function to call on event.
+document.getElementById('killerCard').addEventListener('click', pickKiller);
+
+//If pickKiller function is excecuted/card is clicked then excecute this function:
+const pickWeapon = () => {
+  if (killerIsPicked && !weaponIsPicked) {
+    document.getElementById('weaponLoader').style.visibility='visible'
+    setTimeout(function () {
+      document.getElementById('weaponLoader').style.visibility='hidden'
+      mystery.weapon = randomSelector(weapons); //Randomly select a weapon from weapons and add it to the mystery object
+      document.getElementById('weaponCard').style.background = mystery.weapon.color
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`
+  document.getElementById('weaponWeight').innerHTML = `Weight: ${mystery.weapon.weight}`
+  document.getElementById('weaponFlexibility').innerHTML = `Flexibility: ${mystery.weapon.flexibility}`
+  document.getElementById('weaponImage').src = `${mystery.weapon.image}`
+  weaponIsPicked = true
+    }, 2500)
+  }
+};
+document.getElementById('weaponCard').addEventListener('click', pickWeapon);
+
+//If pickKiller AND pickWeapon function is excecuted/cards are clicked then excecute this function on the pickroom:
+const pickRoom = () => {
+  if (killerIsPicked && weaponIsPicked) {
```

Comment:
Do you want to be able to pick the room again if it's already picked?
`  if (killerIsPicked && weaponIsPicked && roomIsPicked) {`

Reply:
Yes! Good point. Thank you! I noticed that the room-card was clickable also after first click although it shouldn't have been.